source/tutorials/fmt-cdiff.rst

```diff
@@ -0,0 +1,478 @@
+
+.. question::
+   View the ``table.qzv`` QIIME 2 artifact, and in particular the *Interactive Sample Detail* tab in that visualization. What value would you choose to pass for ``--p-sampling-depth``? How many samples will be excluded from your analysis based on this choice? How many total sequences will you be analyzing in the ``core-metrics-phylogenetic`` command?
+
+.. command-block::
+
+   qiime diversity core-metrics-phylogenetic \
+     --i-phylogeny rooted-tree.qza \
+     --i-table table.qza \
+     --p-sampling-depth 1000 \
+     --m-metadata-file sample-metadata.tsv \
+     --output-dir core-metrics-results
+
+Here we set the ``--p-sampling-depth`` parameter to 1109. This value was chosen based on the number of sequences in the ``L3S341`` sample because it's close to the number of sequences in the next few samples that have higher sequence counts, and because it is considerably higher (relatively) than the number of sequences in the one sample that has fewer sequences. This will allow us to retain most of our samples. The one sample that has fewer sequences will be dropped from the ``core-metrics-phylogenetic`` analyses and anything that uses these results.
+
+.. note:: The sampling depth of 1109 was chosen based on the DADA2 feature table summary. If you are using a Deblur feature table rather than a DADA2 feature table, you might want to choose a different even sampling depth. Apply the logic from the previous paragraph to help you choose an even sampling depth.
+
+.. note:: In many Illumina runs you'll observe a few samples that have very low sequence counts. You will typically want to exclude those from the analysis by choosing a larger value for the sampling depth at this stage.
+
+After computing diversity metrics, we can begin to explore the microbial composition of the samples in the context of the sample metadata. This information is present in the `sample metadata`_ file that was downloaded earlier.
+
+We'll first test for associations between categorical metadata columns and alpha diversity data. We'll do that here for the Faith Phylogenetic Diversity (a measure of community richness) and evenness metrics.
+
+.. command-block::
+
+   qiime diversity alpha-group-significance \
+     --i-alpha-diversity core-metrics-results/faith_pd_vector.qza \
+     --m-metadata-file sample-metadata.tsv \
+     --o-visualization core-metrics-results/faith-pd-group-significance.qzv
+
+   qiime diversity alpha-group-significance \
```

Comment:
This command fails for me:

```bash
qiime diversity alpha-group-significance \
  --i-alpha-diversity core-metrics-results/evenness_vector.qza \
  --m-metadata-file sample-metadata.tsv \
  --o-visualization core-metrics-results/evenness-group-significance.qzv \
  --verbose
Traceback (most recent call last):
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.5/site-packages/q2cli/commands.py", line 274, in __call__
    results = action(**arguments)
  File "<decorator-gen-379>", line 2, in alpha_group_significance
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.5/site-packages/qiime2/sdk/action.py", line 231, in bound_callable
    output_types, provenance)
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.5/site-packages/qiime2/sdk/action.py", line 424, in _callable_executor_
    ret_val = self._callable(output_dir=temp_dir, **view_args)
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.5/site-packages/q2_diversity/_alpha/_visualizer.py", line 101, in alpha_group_significance
    groups[j])
  File "/Users/matthew/.conda/envs/q2dev/lib/python3.5/site-packages/scipy/stats/mstats_basic.py", line 1058, in kruskal
    T = 1. - np.sum(v*(k**3-k) for (k,v) in iteritems(ties))/float(ntot**3-ntot)
ZeroDivisionError: float division by zero

Plugin error from diversity:

  float division by zero

See above for debug info.
```

We might need to open up an issue on q2-diversity for this, too.

Reply:
Adding to list ...